src/test/getarg_tests.cpp

```diff
@@ -60,18 +60,18 @@ BOOST_AUTO_TEST_CASE(boolarg)
     BOOST_CHECK(!GetBoolArg("-foo", false));
     BOOST_CHECK(!GetBoolArg("-foo", true));
 
-    ResetArgs("-foo -nofoo");  // -foo should win
-    BOOST_CHECK(GetBoolArg("-foo", false));
-    BOOST_CHECK(GetBoolArg("-foo", true));
-
-    ResetArgs("-foo=1 -nofoo=1");  // -foo should win
-    BOOST_CHECK(GetBoolArg("-foo", false));
-    BOOST_CHECK(GetBoolArg("-foo", true));
+    ResetArgs("-foo -nofoo");  // -nofoo should win
+    BOOST_CHECK(!GetBoolArg("-foo", false));
+    BOOST_CHECK(!GetBoolArg("-foo", true));
```

Comment:
Looks like the duplicates were introduced when the `-nosomething` feature was introduced, and there are several others. Do you want them all cleaned up?

Reply:
Actually, I'll open a ticket for this, because changing it now would cause significant merge conflicts when bitcoin/bitcoin#10607 is pulled in (as part of the `mapArgs` changes this PR is preparing for).

Reply:
Opened #3121.